test/mimemagic_test.rb

```diff
@@ -113,20 +113,4 @@
     MimeMagic.by_magic(StringIO.new 'Y MAGICTEST').should.equal 'application/mimemagic-test'
     MimeMagic.by_magic(StringIO.new 'Z MAGICTEST').should.equal nil
   end
-
-  it 'should handle different file objects' do
-    MimeMagic.add('application/mimemagic-test', magic: [[0, 'MAGICTEST']])
-    class ReadableObj
-      def read
-        'MAGICTEST'
-      end
-    end
-    MimeMagic.by_magic(ReadableObj.new).should.equal 'application/mimemagic-test'
-    class StringableObject
-      def to_s
-        'MAGICTEST'
-      end
-    end
-    MimeMagic.by_magic(StringableObject.new).should.equal 'application/mimemagic-test'
-  end
```

Comment:
You're right, there was no reason, I added them back, but changed `ReadableObj` to `IOObject`. Even though MimeMagic currently uses only `#read` and `#rewind`, it might use other IO methods in the future, and I think it's good to set up this "IO contract" upfront.